libs/store_helper/ebpf_store_helper.c

```diff
@@ -11,25 +11,34 @@
 #include "ebpf_store_helper.h"
 #include "ebpf_windows.h"
 
-uint32_t
+#ifndef USER_MODE
+#define _EBPF_RESULT(x) (NT_SUCCESS(x) ? EBPF_SUCCESS : EBPF_FAILED)
+#else
+#define _EBPF_RESULT(x) win32_error_code_to_ebpf_result(x)
+#endif
+
+#define IS_SUCCESS(x) (x == EBPF_SUCCESS)
+
+ebpf_result_t
 ebpf_store_open_or_create_provider_registry_key(_Out_ ebpf_registry_key_t* provider_key)
 {
-    __return_type status = _SUCCESS;
+    ebpf_result_t status = EBPF_SUCCESS;
     ebpf_registry_key_t root_key = NULL;
     *provider_key = NULL;
 
     // Open (or create) root eBPF registry path.
 #ifdef USER_MODE
-    status = create_registry_key(ebpf_root_registry_key, EBPF_ROOT_RELATIVE_PATH, REG_CREATE_FLAGS, &root_key);
+    status =
+        _EBPF_RESULT(create_registry_key(ebpf_root_registry_key, EBPF_ROOT_RELATIVE_PATH, REG_CREATE_FLAGS, &root_key));
 #else
-    status = create_registry_key(NULL, EBPF_ROOT_REGISTRY_PATH, REG_CREATE_FLAGS, &root_key);
+    status = _EBPF_RESULT(create_registry_key(NULL, EBPF_ROOT_REGISTRY_PATH, REG_CREATE_FLAGS, &root_key));
 #endif
     if (!IS_SUCCESS(status)) {
         goto Exit;
     }
 
     // Open (or create) program data registry path.
-    status = create_registry_key(root_key, EBPF_PROVIDERS_REGISTRY_PATH, REG_CREATE_FLAGS, provider_key);
+    status = _EBPF_RESULT(create_registry_key(root_key, EBPF_PROVIDERS_REGISTRY_PATH, REG_CREATE_FLAGS, provider_key));
```

Comment:
I'd like to see create_registry_key() changed to return ebpf_result_t

Reply:
Converted all low-level registry functions to return ebpf_result_t